tket/src/Mapping/LexiLabelling.cpp

```diff
@@ -0,0 +1,65 @@
+#include "Mapping/LexiLabelling.hpp"
+
+namespace tket {
+
+bool LabellingRoutingMethod::check_method(
+    const std::shared_ptr<MappingFrontier>& mapping_frontier,
+    const ArchitecturePtr& architecture) const {
+  std::set<Node> already_checked;
+  // *it = {UnitID, {Vertex, Port}}
+  for (auto it = mapping_frontier->quantum_boundary->get<TagKey>().begin();
+       it != mapping_frontier->quantum_boundary->get<TagKey>().end(); ++it) {
+    Edge e0 = mapping_frontier->circuit_.get_nth_out_edge(
+        it->second.first, it->second.second);
+    Vertex v0 = mapping_frontier->circuit_.target(e0);
+    Node node = Node(it->first);
+    // i.e. skip already checked vertices
+    if (already_checked.find(node) == already_checked.end()) {
+      already_checked.insert(node);
+      // for coutner<n_edges check, to avoid iterating through many many qubits
+      int n_edges =
+          mapping_frontier->circuit_.n_in_edges_of_type(v0, EdgeType::Quantum);
+      int counter = 1;  // 1 edge
+      auto jt = it;
+      ++jt;
+      for (; jt != mapping_frontier->quantum_boundary->get<TagKey>().end() &&
```

Comment:
This for loop looks a bit unintuitive, have you considered doing a while loop instead?

Reply:
This started as a while loop, but I failed to get the logic right for it so shifted back. I'll attempt another rewrite.

Reply:
No idea what I was doing wrong before, is now a while loop.